Doc/library/datetime.rst

```diff
@@ -1486,6 +1515,23 @@ In boolean contexts, a :class:`.time` object is always considered to be true.
    error-prone and has been removed in Python 3.5.  See :issue:`13936` for full
    details.
 
+
+Other constructors:
+
+.. classmethod:: time.fromisoformat(date_string)
+
+  Return a :class:`time` corresponding to a *time_string* in one of the ISO 8601
+  formats emitted by :meth:`time.isoformat`. Specifically, this function supports
+  strings in the format(s) ``HH[:MM:[SS[.mmm[mmm[+HH:MM]]]]]]``.
```

Comment:
I think you need to parse the time zone without :MM:SS.mmmmmm, e.g. 09:12:32+11:00. See your _datetime.datetime_ format.